solution-2.js

```diff
@@ -0,0 +1,20 @@
+function debounce(func, delay) {
+    let timeoutId;
+  
+    return function(...args) {
+      clearTimeout(timeoutId);
+      timeoutId = setTimeout(() => func.apply(this, args), delay);
+    };
+}
+
+const logMessage = (message) => {
+      console.log(message);
+};
+
+const debouncedLog = debounce(logMessage, 1000);
+
+debouncedLog("Hello");
+setTimeout(() => debouncedLog("World"), 2000);  // Won't log because it's within 2 seconds
```

Comment:
Too verify if debounce is working properly, You have to call a function repeatedly and it should only execute after specified debounce time.

Reply:
Okay. Noted.